GVFS/GVFS.FunctionalTests/GVFSTestConfig.cs

```diff
@@ -16,14 +17,23 @@ public static class GVFSTestConfig
 
         public static bool ReplaceInboxProjFS { get; set; }
 
+        public static bool StartedFromDebugger { get; set; }
+
         public static string PathToGVFS
         {
             get
             {
-                return
-                    TestGVFSOnPath ? 
-                    Properties.Settings.Default.PathToGVFS : 
-                    Path.Combine(Properties.Settings.Default.CurrentDirectory, Properties.Settings.Default.PathToGVFS);
+                if (TestGVFSOnPath)
+                {
+                    return Properties.Settings.Default.PathToGVFS;
+                }
+
+                if (StartedFromDebugger && RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    return Path.Combine(Properties.Settings.Default.CurrentDirectory, "..", "..", "..", "..", "..", "..", "Publish", Properties.Settings.Default.PathToGVFS);

```

Comment:
What is your goal with this change? Based on the code here, it looks like you still have to remember to publish before you run your tests. So for example, if you make a change and then just build and run, you will run the tests on the _last version you published_, not the version you just built. That seems likely to cause some a lot of time and sanity :-)

Is there any way we can remove the requirement to publish before you can run the tests? That would be ideal because we can run the tests from the debugger, and also publishing takes a non-trivial amount of time and it would be nice to get rid of that while testing changes.

Reply:
> So for example, if you make a change and then just build and run, you will run the tests on the last version you published, not the version you just built. That seems likely to cause someone a lot of time and sanity :-)

Ahh good point, I did not realize that building in VS for Mac would not perform the publish step.

>Is there any way we can remove the requirement to publish before you can run the tests? That would be ideal because we can run the tests from the debugger, and also publishing takes a non-trivial amount of time and it would be nice to get rid of that while testing changes.

I think we might be able to change this so that when started from the debugger we call `dotnet` rather than `gvfs`.  Let me give that a try.

Reply:
> I think we might be able to change this so that when started from the debugger we call dotnet rather than gvfs. Let me give that a try.

@sanoursa after spending a bunch of time digging into it, the approach I suggested above appears to be a real rat's nest.

I think it will be easier\clean to add a post-build step to GVFS.FunctionalTests on Mac that does a `dotnet publish` I'll see if I can get this approach to work.